app/models/manageiq/providers/amazon/agent_coordinator_worker.rb

```diff
@@ -1,17 +1,21 @@
 class ManageIQ::Providers::Amazon::AgentCoordinatorWorker < MiqWorker
-  require_nested :Runner
+  include MiqWorker::ReplicaPerWorker
 
-  include PerEmsWorkerMixin
+  require_nested :Runner
 
   self.required_roles = ['smartproxy']
+  self.workers        = 1
 
-  def self.ems_class
-    ManageIQ::Providers::Amazon::CloudManager
+  def self.has_required_role?
+    super && all_valid_ems_in_zone.any?
   end
 
-  def self.desired_queue_names
-    # All cloud managers will share the same agent coordinator
-    super.any? ? ["ems_agent_coordinator"] : []
+  def self.all_valid_ems_in_zone
+    ems_class.where(:zone_id => MiqServer.my_server.zone.id).select { |e| e.enabled && e.authentication_status_ok? }
+  end
```

Comment:
Maybe some follow-up refactoring but I could see pulling this out into a `ProviderWorkerMixin` since it is in `PerEmsWorkerMixin` and the `PerEmsTypeWorkerMixin` (and this is neither of those)

Reply:
It was also duplicated by the VimBroker, actually the Broker had a lot of similarities to this worker type.  Run only one if there are any valid EMSs of a specific type in the zone.

Reply:
https://github.com/ManageIQ/manageiq/pull/21060